modules/geo-layers/src/global-grid-layer/global-grid-utils/h3-utils.ts

```diff
@@ -0,0 +1,52 @@
+// deck.gl
+// SPDX-License-Identifier: MIT
+// Copyright (c) vis.gl contributors
+
+import {lerp} from '@math.gl/core';
+import {
+  type CoordPair,
+  type H3IndexInput, // Either string or SplitLong
+  cellToBoundary,
+  cellToLatLng
+} from '../h3-js-bigint/h3-js-bigint';
+import {normalizeLongitudes} from './global-grid-utils';
+
+// UTILITIES - TODO can these be generalized using the decoder?
```

Comment:
A lot of this code is here to handle the `coverage` prop. If we make the centroid calculation based purely on the vertices, it will be possible to offer the `coverage` prop for all grid systems